python/ray/data/datasource/file_based_datasource.py

```diff
@@ -305,26 +303,72 @@ def write_block(write_path: str, block: Block):
                     writer_args_fn=write_args_fn,
                     **write_args,
                 )
-
-        write_block = cached_remote_fn(write_block).options(**ray_remote_args)
+            # TODO: decide if we want to return richer object when the task
+            # succeeds.
+            return "ok"
 
         file_format = self._FILE_EXTENSION
         if isinstance(file_format, list):
             file_format = file_format[0]
 
-        write_tasks = []
+        builder = DelegatingBlockBuilder()
+        for block in blocks:
+            builder.add_block(block)
+        block = builder.build()
+
         if not block_path_provider:
             block_path_provider = DefaultBlockWritePathProvider()
-        for block_idx, block in enumerate(blocks):
-            write_path = block_path_provider(
+        write_path = block_path_provider(
+            path,
+            filesystem=filesystem,
+            dataset_uuid=dataset_uuid,
+            block=block,
+            block_index=ctx.task_idx,
+            file_format=file_format,
+        )
+        return write_block(write_path, block)
+
+    @Deprecated
+    def do_write(
+        self,
+        blocks: List[ObjectRef[Block]],
+        metadata: List[BlockMetadata],
+        path: str,
+        dataset_uuid: str,
+        filesystem: Optional["pyarrow.fs.FileSystem"] = None,
+        try_create_dir: bool = True,
+        open_stream_args: Optional[Dict[str, Any]] = None,
+        block_path_provider: BlockWritePathProvider = DefaultBlockWritePathProvider(),
+        write_args_fn: Callable[[], Dict[str, Any]] = lambda: {},
+        _block_udf: Optional[Callable[[Block], Block]] = None,
+        ray_remote_args: Dict[str, Any] = None,
+        **write_args,
+    ) -> List[ObjectRef[WriteResult]]:
+        """Creates and returns write tasks for a file-based datasource."""
```

Comment:
Nothing is calling do_write() any more right? Should we just remove it for our internal datasources?

Reply:
done.